rust_src/src/filelock.rs

```diff
@@ -0,0 +1,46 @@
+//! Lock files for editing.
+
+use remacs_macros::lisp_fn;
+
+use crate::{
+    lisp::LispObject,
+    remacs_sys::{lock_file, unlock_file},
+    remacs_sys::Qstringp,
+    threads::ThreadState,
+};
+
+/// Lock FILE, if current buffer is modified.
+/// FILE defaults to current buffer's visited file,
+/// or else nothing is done if current buffer isn't visiting a file.
+///
+/// If the option `create-lockfiles' is nil, this does nothing.
+#[lisp_fn(min = "0", name = "lock-buffer")]
```

Comment:
Good job handling the min and adding a name. However, the name is not needed here. We only use the `foo_lisp` style naming scheme when there is a conflicting function with the same name as the DEFUN. Is there a `lock_buffer` or `unlock_buffer` function in the code?

Reply:
Dropped from `lock_buffer`.

A public `unlock_buffer`, however, [exists](https://github.com/remacs/remacs/blob/11d0cb791daf6a440599c4ab1f9b1bb2db8b3b2d/src/filelock.c#L792).